### ISSUE TITLE
Cursor auto-consumes first 10 results.

### ISSUE THREAD

**Issue by JP-Dhabolt** [arangodb/arangodb#2257]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [x] 3.0
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [x] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [x] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

I'm issuing AQL via:
- [ ] web interface with this browser:     running on this OS:
- [x] arangosh
- [ ] this Driver:

The following problem occurs:

When creating a cursor to consume programatically, the first 10 results are lost.  This can be seen in the following:
`stmt = db._createStatement({ "query": "FOR i IN 1..11 RETURN i"});`
`testCursor = stmt.execute();`
`testCursor.next();`
This first call to the `testCursor.next()` function returns 11.

Instead I would be expecting: 

The first call to the `testCursor.next()` function returns 1.

Currently, the first 10 results can only be consumed programatically by calling the `testCursor.toArray()` function, or examining `testCursor.data.result`.

**Comment by jsteemann** [comment]
When executing the above code programmatically (i.e. not in interactive shell mode), the result is as expected. The problem will only occur in arangosh when the commands are run individually. The command `testCursor = stmt.execute();` executes the statement and sets the cursor at the first result value. However, as the shell always prints the result of the last executed expression, it will call the `toString` function for the cursor. This fetches the first 10 elements and prints them, which will forward the cursor.
In non-interactive mode, no `toString` function of the cursor will be called. The result is also correct when all commands are put on a single line (because `toString` will not be called then), i.e.
```
stmt = db._createStatement({ "query": "FOR i IN 1..11 RETURN i"}); testCursor = stmt.execute(); testCursor.next();
```

**Comment by fceller** [comment]
Alternatively you can put a "var" in front of the

**Comment by JP-Dhabolt** [comment]
Frank, thanks for that.  I was re-using the variable during my testing, and didn't realize prefixing var would suppress the output.

Jan, understanding why this is occurring is helpful.  If the plan is to leave things as they are, the documentation should be updated to represent this, as I would expect code to work deterministically between interactive and non-interactive modes.  I know I personally use the interactive mode for prototyping, and would expect the modes to work the same unless stated otherwise.

**Comment by baslr** [comment]
@JP-Dhabolt, the autoconsuming fix is now in devel.

**Comment by jsteemann** [comment]
The issue has been fixed in devel and will become part of a future release.
Thanks @baslr for the initial patch!

**Comment by dothebart** [comment]
The fix is available with the new 3.2 Alpha 3 for testing. Closing now.
